.ci/magician/cmd/collect_nightly_test_status.go

```diff
@@ -0,0 +1,213 @@
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+package cmd
+
+import (
+	"fmt"
+	"magician/cloudstorage"
+	"magician/provider"
+	"magician/teamcity"
+	utils "magician/utility"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	NIGHTLY_DATA_BUCKET = "nightly-test-data"
+)
+
+var cntsRequiredEnvironmentVariables = [...]string{
+	"TEAMCITY_TOKEN",
+}
+
+type TestInfo struct {
+	Name         string `json:"name"`
+	Status       string `json:"status"`
+	Service      string `json:"service"`
+	ErrorMessage string `json:"error_message"`
+	LogLink      string `json"log_link`
+}
+
+// collectNightlyTestStatusCmd represents the collectNightlyTestStatus command
+var collectNightlyTestStatusCmd = &cobra.Command{
+	Use:   "collect-nightly-test-status",
+	Short: "Collects and stores nightly test status",
+	Long: `This command collects nightly test status, stores the data in JSON files and upload the files to GCS.
+
+
+	The command expects the following argument(s):
+	1. Custom test date in YYYY-MM-DD format. default: ""(current time when the job is executed)
+
+	It then performs the following operations:
+	1. Collects nightly test status of the execution day or the specified test date (if provided)
+	2. Stores the collected data in JSON files
+	3. Uploads the JSON files to GCS
+
+	The following environment variables are required:
+` + listCNTSRequiredEnvironmentVariables(),
+	Args: cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		env := make(map[string]string)
+		for _, ev := range cntsRequiredEnvironmentVariables {
+			val, ok := os.LookupEnv(ev)
+			if !ok {
+				return fmt.Errorf("did not provide %s environment variable", ev)
+			}
+			env[ev] = val
+		}
+
+		tc := teamcity.NewClient(env["TEAMCITY_TOKEN"])
+		gcs := cloudstorage.NewClient()
+
+		date := time.Now()
+		customDate := args[0]
+		//
+		if customDate != "" {
+			parsedDate, err := time.Parse("2006-01-02", customDate) // input format YYYY-MM-DD
+			// Set the time to 6pm PT
+			date = time.Date(parsedDate.Year(), parsedDate.Month(), parsedDate.Day(), 18, 0, 0, 0, parsedDate.Location())
+			if err != nil {
+				return fmt.Errorf("invalid input time format: %w", err)
+			}
+		}
+
+		return execCollectNightlyTestStatus(date, tc, gcs)
+	},
+}
+
+func listCNTSRequiredEnvironmentVariables() string {
+	var result string
+	for i, ev := range cntsRequiredEnvironmentVariables {
+		result += fmt.Sprintf("\t%2d. %s\n", i+1, ev)
+	}
+	return result
+}
```

Comment:
This makes sense for debugging but we probably don't want to put this in the logs long-term.

```suggestion
```

Reply:
removed.

Reply:
I totally misread this as logging the value of the env var, not the name 🤦 please feel free to add it back, sorry about that!

Reply:
lol, I thought you mean we don't need to log the names as well, which, I thought, also makes sense. I'll add it back though